### DISCUSSION TITLE
"Enter REST" action via a MIDI Control Change code during partition entry

### DISCUSSION THREAD

**Discussion by zr-jl** [musescore/MuseScore#20750] [Ideas]
### Proposal for a new desired action in addition to the actions in MuseScore's MIDI remote control list.

Hello,
Entering scores (voices for choirs in my case) with MuseScore 
**via MIDI messages sent from a tablet touchscreen 
is, in practice, a method that I find simple, fast and efficient.** 
(for software configuration details, see the links at the bottom of the message)

In addition to classic MIDI messages such as Note ON or Note OFF, 
the actions proposed in the "MIDI Control" window with MuseScore's "Preferences", 
which are useful for facilitating note entry, work perfectly:
* Toggle note input mode
* Notes durations (7)
* Toggle duration dot and dot: double
 (see complete list of actions in screenshot below)

### However, the ability to enter a rest via MIDI Control Change is sorely lacking.

Entering a rest, as you do with the number "0" (zero) from the computer keyboard 
would also be very useful in the list of proposed actions (see list below)
and could be integrated into the MIDI scripts of the StreamDeck MIDI plugin.

For your information, the closest existing MIDI remote control action is : 
**"Toggle rest"** 
does not react like the "0" on the computer keyboard, but only allows you, if you're not in note entry mode and one or more notes are selected, to replace them with a silence of the same duration. 

### For those interested in this alternative input method, here's some additional information :

**Details informations** 
A) Preparing the software configuration:
1. Install Elgato StreamDeck Mobile App
https://apps.apple.com/us/app/elgato-stream-deck-mobile/id1440014184

2. Add and configure the MIDI Plugin developed by Gunnar Carlson for this App
https://trevligaspel.se/streamdeck/midi/index.html

3. Configure your profiles with Stream Deck App... (a few hours' work)  and then 

B) Preparing for song entry:

### 1. On your tablet choose the voice to be entered and the key of the piece (here Tenor voice and piece in Eb / Cm ) 

![Choose voice and key](https://github.com/musescore/MuseScore/assets/126709343/0a1aee7b-e12c-4fbc-a1ea-c556b7b41b69)

### with 2 StreamDeck App-sessions in iPad Pro: SIMPLY ONE NOTE, ONE KEY-Touch

![One note one touch](https://github.com/musescore/MuseScore/assets/126709343/d7275796-6f3e-4cfe-961a-d7c6b84c31e1)

List of possible actions for controlling MuseScore 4.2 remotely by sending MIDI Control Change.
for mac see MuseScore Preferences : cmd + ,  (or via MuseScore menu list)
then open MIDI mappings
<img width="544" alt="MIDI Mappings" src="https://github.com/musescore/MuseScore/assets/126709343/45527e6f-d009-40b6-9e3b-0857693583c1">
![StreamDeck MIDI Plugin ](https://github.com/musescore/MuseScore/assets/126709343/c411a166-3336-46f6-b1d6-c7c74bcb2572)

**Comment by cbjeukendrup** [comment]
I logged a simple and easy-to-read feature request about this here: https://github.com/musescore/MuseScore/issues/20998

**Reply by zr-jl** [reply]
Thank you very much for this service

I'm gradually discovering new ways of doing things

I'm not used to using Github yet
but I'm learning

**Reply by s11141827** [reply]
Well so am I
